tests/functional/UseCase/Bucket/ShareLinkBucketTest.php

```diff
@@ -0,0 +1,359 @@
+        $path = new RepeatedField(GPBType::STRING);
+        $path[] = $bucketDatabaseName;
+        $columns = new RepeatedField(GPBType::MESSAGE, CreateTableCommand\TableColumn::class);
+        $columns[] = (new CreateTableCommand\TableColumn())
+            ->setName('ID')
+            ->setType(Bigquery::TYPE_INTEGER);
+        $command = (new CreateTableCommand())
+            ->setPath($path)
+            ->setTableName('TESTTABLE_BEFORE')
+            ->setColumns($columns);
+        $handler(
+            $this->sourceProjectCredentials,
+            $command,
+            []
+        );
+        $sourceBqClient->runQuery($sourceBqClient->query(sprintf(
+            'INSERT INTO %s.`TESTTABLE_BEFORE` (`ID`) VALUES (1)',
+            BigqueryQuote::quoteSingleIdentifier($bucketDatabaseName)
+        )));
+
+        $targetProjectBqClient = $this->clientManager->getBigQueryClient($this->targetProjectCredentials);
+
+//      check that the Project2 cannot access the table yet
+        $dataset = $targetProjectBqClient->dataset($linkedBucketSchemaName);
+        $this->assertFalse($dataset->exists());
+
+        $publicPart = (array) json_decode(
+            $this->sourceProjectResponse->getProjectUserName(),
+            true,
+            512,
+            JSON_THROW_ON_ERROR
+        );
+        /** @var string $sourceProjectId */
+        $sourceProjectId = $publicPart['project_id'];
+        // share the bucket
+        $handler = new ShareBucketHandler($this->clientManager);
+        $command = (new ShareBucketCommand())
+            ->setSourceProjectId($sourceProjectId)
+            ->setSourceBucketObjectName($bucketDatabaseName)
+            ->setSourceProjectReadOnlyRoleName($this->sourceProjectResponse->getProjectReadOnlyRoleName());
+
+        $result = $handler(
+            $this->getCredentials(),
+            $command,
+            []
+        );
+
+        assert($result instanceof ShareBucketResponse);
+        $listing = $result->getBucketShareRoleName();
+
+        $publicPart = (array) json_decode(
+            $this->targetProjectResponse->getProjectUserName(),
+            true,
+            512,
+            JSON_THROW_ON_ERROR
+        );
+        /** @var string $targetProjectId */
+        $targetProjectId = $publicPart['project_id'];
+        // link the bucket
+        $handler = new LinkBucketHandler($this->clientManager);
+        $command = (new LinkBucketCommand())
+            ->setStackPrefix($this->getStackPrefix())
+            ->setTargetProjectId($targetProjectId)
+            ->setTargetBucketId($linkedBucketSchemaName)
+            ->setSourceShareRoleName($listing); // listing
+
+        // it is sourceProject who does the grants -> that's why the sourceProjectCredentials
+        $result = $handler(
+            $this->getCredentials(),
+            $command,
+            []
+        );
+
+        assert($result instanceof LinkedBucketResponse);
```

Comment:
taktez tady: $this->assertInstanceOf a zkusit assertnout jmeno toho noveho schematu. nebo aspon ze to neni prazdny nebo tak neco